hazelcast-cloud/src/test/java/com/hazelcast/aws/security/EC2RequestSignerTest.java

```diff
@@ -47,6 +50,35 @@
 
 
     @Test
+    public void test_Constasts(){
+        Constants constants = new Constants();
+        assertNotNull(constants.DATE_FORMAT);
+        assertNotNull(constants.DOC_VERSION);
+        assertNotNull(constants.SIGNATURE_METHOD_V4);
+        assertNotNull(constants.GET);
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenConfigIsNull(){
+        new EC2RequestSigner(null,"","");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeStampIsNull(){
+        new EC2RequestSigner(new AwsConfig(),null,"");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeSignServiceIsNull(){
+        new EC2RequestSigner(new AwsConfig(),"","").sign(null,new HashMap<String, String>());
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void testEC2RequestSigner_whenTimeSignAttributeIsNull(){
+        new EC2RequestSigner(new AwsConfig(),"","").sign("",null);
+    }
+    
+    @Test
```

Comment:
can you check `deriveSigningKeyTest` again? It has two big try/catch block that seems unrelated to each other? This test can be simplified.